app/src/main/java/io/nwhacks/nfc/WriteFragment.java

```diff
@@ -101,6 +104,14 @@ public void onEvent(@Nullable DocumentSnapshot documentSnapshot, @Nullable Fireb
                     di = documentSnapshot.toObject(DeviceInfo.class);
                     writeId.setText(di.writeId);
                     writeName.setText(di.writeName);
+                    if (di.writeApplicantType != null){
+                        applicantCollection = ApplicantInfo.applicantMap.get(di.writeApplicantType);
+                        writeType.setText(di.writeApplicantType);
+                        setColor(DEFAULT_COLOR);
+                    }else{
+                        MainActivity.toast(getContext(), "No ApplicantType for this applicant. Please check that you've selected an applicant.");
```

Comment:
Great, this covers the missing record case. Can we deal with the case that the second record does exist, but doesn't contain a valid `applicantType` (i.e. tag was tampered with)? You can either detach this else case and make it execute if `applicantCollection == null`, or write a separate if statement after this conditional block.

Reply:
I decided to change the larger if statement checking `records.size() > 0` to `applicantType != null`

I've also moved the if statement to the end of this larger if statement.